tests/test_pointcompare.py

```diff
@@ -67,6 +67,18 @@ def test_skill_from_observation_with_missing_values(modelresult_oresund_2d):
     assert not np.any(np.isnan(s))
 
 
+def test_extraction_no_overlap(modelresult_oresund_2d):
+    o1 = PointObservation(
+        "tests/testdata/smhi_2095_klagshamn_shifted.dfs0",
+        x=366844,
+        y=6154291,
+        name="Klagshamn",
+    )
+    mr = modelresult_oresund_2d
+    mr.add_observation(o1, item=0)
+    mr.extract()
```

Comment:
It is not clear what this test is asserting in the case of no overlap.

If there is no overlap (time or space) then it is not possible to calculate a residual and no skill.

Should an exception be raised, or only if some flag is set, the behaviour is not clear from reading this test.

Reply:
I guess @Hendrik1987 made this to illustrate that it gave an error. But you're right we need to add a test with the desired behaviour. 

I have been wondering what the right behaviour is in case of no data. I see these options when constructing a Comparer with no data:
* raise an error
* return None and...
   - do nothing 
   - issue a warning

I decided to return None and issue a warning. I imagine that if you loop through 100s of observations in a script you don't want it to stop just because one observation has no entries in a given period. That is common and would be annoying. 

But what about the ComparerCollection? mr.extract() returns a ComparerCollection... what should we do if all extractions are None - should the ComparerCollection be None also or should it be a collection with zero-length (`c.n_comparers == 0`). So far I've chosen the latter.

Reply:
The assert statement has now been added. And the user would see this:

    UserWarning: No time overlap between model result and observation!
    UserWarning: Could not add observation

Reply:
Now also tests for UserWarnings

Reply:
@ecomodeller do you agree with the behaviour (i.e. returning None and issuing UserWarning)?